### PR TITLE
gobject-introspection 1.78.1: Rebuild for win-64

### PR DESCRIPTION
**Pull request by skupr-anaconda**
**Destination channel:** main

### Links

- [PKG-5392](https://anaconda.atlassian.net/browse/PKG-5392) 
- [Upstream repository](https://gitlab.gnome.org/GNOME/gobject-introspection)

### Explanation of changes:

- Bump the build number to `1`
- Fix caret  `^` and double quotes `"` symbols in `bld.bat.` It allows correctly taking the flag `--prefix=%LIBRARY_PREFIX%`. Before that `pkg-config` couldn't locate `*.pc` files of `cairo`, `glib`, etc. 
- Add `-Ddoctool=enabled` flag on win for better log messages. It requires `mako` and `markdown` python packages in `host`.
- Apply the upstream patch https://gitlab.gnome.org/GNOME/gobject-introspection/-/merge_requests/474. Please take a look at the explanation in th **Notes** below.
- Remove  `downstreams: - pygobject` as we do not have this package on defaults

### Notes:

- A downstream package `librsvg` fails on win64 with an error:
`C:\b\abs_9dv6t0n677\croot\librsvg_1721748049684\_h_env\Library\include\glib-2.0\glib\gstring.h:193: syntax error, unexpected identifier in '__forceinline' at '__forceinline'`

It is caused by the `glib` library. 
`glib`'s `G_ALWAYS_INLINE` since **2.74** expands to the GNU C `always_inline` or MSVC `__forceinline` function attribute depending on the compiler. See https://docs.gtk.org/glib/macros.html and https://gitlab.gnome.org/GNOME/gobject-introspection/-/merge_requests/474/diffs?commit_id=f3c17a6bff87b720a8e08e1eef25d2b53e61e0bc

That's why we rebuild `gobject-introspection` with a patch

[PKG-5392]: https://anaconda.atlassian.net/browse/PKG-5392?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/handle_visual_studio_forceinline.patch [hunk 1]
```diff
@@ -0,0 +1,24 @@
+From ceca435d0045e1042f4045a0c43e98d7d7e69df8 Mon Sep 17 00:00:00 2001
+From: Serhii Kupriienko
```
Comment (JeanChristopheMorinPerso) [inline recipe/handle_visual_studio_forceinline.patch]: This patch is not from you. Please give proper attributions. You can use https://gitlab.gnome.org/GNOME/gobject-introspection/-/merge_requests/474.patch to get the full patch.

Comment (skupr-anaconda) [reply recipe/handle_visual_studio_forceinline.patch]: Thanks. Will do

**Comment by skupr-anaconda** [comment]
I've noticed that a few platforms _randomly_ can fail with different error messages, for example, because `pkg-config` couldn't locate headers (`cairo.h`) with an additional error `FAILED: tests/scanner/Typedefs-1.0.gir`. 
Reopening the PR solves this issue.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -45,14 +51,15 @@ requirements:
     - {{ cdt('xorg-x11-proto-devel') }}  # [linux]
     - {{ cdt('libxau-devel') }}          # [linux]
     - {{ cdt('libxext-devel') }}         # [linux]
-    - {{ posix }}patch
-    - {{ posix }}sed
   host:
-    - pkg-config
+    - pkg-config 0.29
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]: I don't think it's a good idea to pin pkg-config.

File: recipe/bld.bat [hunk 3]
```diff
@@ -5,17 +5,18 @@ setlocal EnableDelayedExpansion
...
 :: meson options
 :: (set pkg_config_path so deps in host env can be found)
-set ^"MESON_OPTIONS=^
-  --prefix="%LIBRARY_PREFIX%" ^
+set MESON_OPTIONS=^
+  --prefix=%LIBRARY_PREFIX% ^
   --wrap-mode=nofallback ^
   --buildtype=release ^
   --backend=ninja ^
+  -Ddoctool=enabled ^
```
Comment (JeanChristopheMorinPerso) [inline recipe/bld.bat]: Does it fail if doctool is disabled?

Comment (skupr-anaconda) [reply recipe/bld.bat]: No, it works without it. I enabled it to debug build logs and catch the root cause. MSYS2 enabled it https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-gobject-introspection/PKGBUILD#L54

Comment (JeanChristopheMorinPerso) [reply recipe/bld.bat]: Ok, so I guess we can disable it?

Comment (skupr-anaconda) [reply recipe/bld.bat]: Removed

**Review by JeanChristopheMorinPerso — APPROVED** [review]

**Review by cbouss — APPROVED** [review]

**Pull request merged**